### DISCUSSION TITLE
Save route with waypoints

### DISCUSSION THREAD

**Discussion by ThomasMeister123** [osmandapp/OsmAnd#14261] [Ideas]
Please make it possible to save a route with different waypoints. If you load this saved route, then all Waypoints should be in the waypoint-list (and NOT the track).

**Comment by ThomasMeister123** [comment]
Is it now possible in the newest version? It doesnt look so 😔

**Reply by dmpr0** [reply]
Hello!
GPX navigation doesn't support waypoints, only an additional destination point. You need to add points you want to visit while planning the route or use regular navigation which supports waypoints.

**Reply by ThomasMeister123** [reply]
I dont speak from GPX. I will plan a route in OSMand. And then I will save the route (NOT a track! Just the Routing-/Waypoints!) so I can open it later or transfer it to another Mobile with OSMand. GPX, XML, TXT, ... whatever...

**Reply by vshcherb** [reply]
Yes, we plan to allow to add wpt points during editing the track in Plan Route functionality. Right now you can add them via Context Menu add Track Waypoint

**Reply by ThomasMeister123** [reply]
Sorry, I dont get it. I will plan a trip. From A via B and C to D. Now I will save this planned trip (NOT a GPX-Track! Just the Points ABCD!) I will re-open this trip later, maybe on an other device. This is possible now?
Even more better: Give the opportunity to plan a route on PC (e.g. just a textfile), than transfer this file to the mobile an Import the route (Points!) with one click in Osmand  and THEN calculate the track within OSMand. So you can use different profiles. When you save a route today, it saves the GPX-Track! So I cant use another profile when I reopen it

**Reply by vshcherb** [reply]
You can change profile in Plan Route later, intermediate points are saved as route points so it it's not interoperable but it gives enough flexibility I believe. Though as you mentionned you can't save a route and open exactly, in that case you can share points and build the same route for example.

**Reply by ThomasMeister123** [reply]
OK, but is no solution for me. I plan many routes with waypoints before a holidaytrip. And I will have the flexiibility to delete waypoints later. That is not possible with OSMand because it stores the track and not the waypoints. Then I have to use Mapfactor Navigator for this... 😒

**Reply by vshcherb** [reply]
Navigation by Saved GPX should be identical though Intermediate points are not restored we probably have an open issue for that.

**Reply by ThomasMeister123** [reply]
OK. Thank you for the explanation! Its the same, as long as you dont want delete one of these waypoints. Or change the profile (from bike to  but GPX uses a bike-route.
So, I'm looking forward for this feature :)
